runtime/recipe/view.js

```diff
@@ -81,12 +83,18 @@ class View {
     this._fate = fate;
   }
   get originalFate() { return this._originalFate || "?"; }
+  get originalId() { return this._originalId; }
   get recipe() { return this._recipe; }
   get tags() { return this._tags; } // only tags owned by the view
   set tags(tags) { this._tags = tags; }
   get type() { return this._type; } // nullable
   get id() { return this._id; }
-  set id(id) { this._id = id; }
+  set id(id) {
+    if (!this._originalId) {
+      this._originalId = this._id;
```

Comment:
Which strategies can change the id?

Reply:
When view fate is "copy", during the recipe instantiation a new view is created and assigned a new ID:
https://github.com/PolymerLabs/arcs/blob/master/runtime/arc.js#L228
with this change, the originalId will keep the ID of the view this one was copied from.

Reply:
Ahh I see. That does seem important to keep around.

Reply:
I know we talked about making a more generic solution for this (back when i added original fate). But for now I just keep piling up hacks and todos...